packages/calcite-components/src/components/text-area/text-area.tsx

```diff
@@ -152,6 +152,27 @@ export class TextArea
   /** Specifies the validation icon to display under the component. */
   @Prop({ reflect: true }) validationIcon: string | boolean;
 
+  /**
+   * The current validation state of the component.
+   *
+   * @readonly
+   * @mdn [ValidityState](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState)
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated in form util when syncing hidden input
+  @Prop({ mutable: true }) validity = {
```

Comment:
Can you define a type for this in `src/components/interfaces.ts` and update props?

Reply:
Note: I used the MutableValidtyState from the form util instead of creating another one in interfaces, if that's okay